app/src/ingest_policy_pdfs.py

```diff
@@ -14,17 +20,114 @@
 
 
 def _ingest_policy_pdfs(
-    db_session: db.Session,
     pdf_file_dir: str,
     doc_attribs: dict[str, str],
 ) -> None:
     file_list = get_files(pdf_file_dir)
     embedding_model = app_config.sentence_transformer
+    db_session = app_config.db_session()
     for file in file_list:
         if file.endswith(".pdf"):
-            logger.info(
-                f"Processing pdf file: {file} at {pdf_file_dir} using {embedding_model}, {db_session}, with {doc_attribs}"
+            with smart_open_file(file, "rb") as fin:
+                output_string = extract_text(fin, laparams=LAParams())
+                logger.info(
+                    f"Processing pdf file: {file} at {pdf_file_dir} using {embedding_model}, {db_session}, with {doc_attribs}"
+                )
+
+                parse_pdf_and_add_to_db(
+                    contents=output_string, doc_attribs=doc_attribs, db_session=db_session
+                )
+
+
+def parse_pdf_and_add_to_db(
+    contents: str, doc_attribs: dict[str, str], db_session: db.Session
+) -> None:
+    # Match header in BEM manual
+    header_pattern = r"(BEM\s\d*\s+\d+\sof\s\d+\s+\w.*)"
+    text_split_by_header = re.split(header_pattern, contents)
+    current_title = ""
+    body_content = ""
+    start_new_section = True
+    for text_contents in text_split_by_header:
+        is_header, contents, start_new_section = get_header_and_is_current_section(
+            text_contents, start_new_section
+        )
+        # Check if we need to start a new section
+        if is_header and start_new_section and body_content != "":
+            current_title = contents
+        else:
+            body_content += f"{contents}\n"
+
+    document = Document(name=current_title, content=body_content, **doc_attribs)
+    db_session.add(document)
+
+    process_chunk(body_content, document, db_session)
+    db_session.commit()
```

Comment:
Note that `db_session.commit()` is being called once per file. Have you tested this function on multiple files?
I thought `db_session.commit()` should be called once per `db_session`:
```
    with app_config.db_session() as db_session:
        ingestion_call(pdf_file_dir, doc_attribs)
        db_session.commit()
```

Reply:
@ccheng26 Did you check if you can call `db_session.commit()` multiple times for the same `db_session`?

Reply:
Update: multiple `commit()` calls work. we suspect new transactions are created for subsequent `commit()` calls.